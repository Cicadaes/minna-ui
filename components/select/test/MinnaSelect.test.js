'use strict';

const MinnaSelect = require('../src/MinnaSelect.html');

const selectOpts = [
  { id: 'au', text: 'Australia' },
  { id: 'cn', text: 'China' },
  { id: 'jp', text: 'Japan' },
  { id: 'kr', text: 'Korea' },
  { id: 'other', text: 'Other / Unknown' },
];

describe('MinnaSelect component', () => {
  it('should render correctly', () => {
    const target = document.createElement('div');
    new MinnaSelect({
      target,
      data: {
        options: selectOpts,
      },
    });
    expect(target.innerHTML).toMatchSnapshot();
  });
});
