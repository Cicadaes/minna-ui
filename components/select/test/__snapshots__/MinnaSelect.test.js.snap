// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinnaSelect component should render correctly 1`] = `
"<div class=\\"pos-r dif f-col  \\"><input id=\\"undefined\\" class=\\"select\\" tabindex=\\"0\\" placeholder=\\"Choose...\\" readonly=\\"\\" aria-haspopup=\\"listbox\\">
  <div class=\\"select-caret\\"></div>

  <div class=\\"select-listbox hide\\" role=\\"listbox\\"><div class=\\"option  option-active\\" value=\\"au\\" role=\\"option\\">Australia</div><div class=\\"option  \\" value=\\"jp\\" role=\\"option\\">Japan</div><div class=\\"option  \\" value=\\"kr\\" role=\\"option\\">Korea</div><div class=\\"option  \\" value=\\"other\\" role=\\"option\\">Other / Unknown</div>
    </div></div>"
`;
